WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt

```diff
@@ -77,6 +79,15 @@ class HelpActivity : AppCompatActivity() {
             }
             AnalyticsTracker.track(Stat.SUPPORT_IDENTITY_FORM_VIEWED)
         }
+
+        if (savedInstanceState != null) {
+            mTicketsBeingShown = savedInstanceState.getBoolean(HelpActivity.MY_TICKETS_SHOWN_KEY)
+        }
+
+        if (originFromExtras == Origin.ZENDESK_NOTIFICATION && !mTicketsBeingShown) {
```

Comment:
It might be just me, but I find this solution unnecessarily complicated and a little bit confusing. I do understand what's going on, but not everyone would be familiar with what this is solving, at least not without comments. May I suggest updating the solution as such:

```
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt b/WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt
index 787779cc2b..a7d2f38ee6 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt
@@ -38,8 +38,6 @@ class HelpActivity : AppCompatActivity() {
         intent.extras?.get(WordPress.SITE) as SiteModel?
     }
 
-    private var mTicketsBeingShown: Boolean = false
-
     override fun attachBaseContext(newBase: Context) {
         super.attachBaseContext(LocaleManager.setLocale(newBase))
     }
@@ -80,12 +78,13 @@ class HelpActivity : AppCompatActivity() {
             AnalyticsTracker.track(Stat.SUPPORT_IDENTITY_FORM_VIEWED)
         }
 
-        if (savedInstanceState != null) {
-            mTicketsBeingShown = savedInstanceState.getBoolean(HelpActivity.MY_TICKETS_SHOWN_KEY)
-        }
-
-        if (originFromExtras == Origin.ZENDESK_NOTIFICATION && !mTicketsBeingShown) {
-            mTicketsBeingShown = true
+        /**
+         * If the user taps on a Zendesk notification, we want to show them the `My Tickets` page. However, this
+         * should only be triggered when the activity is first created, otherwise if the user comes back from
+         * `My Tickets` and rotates the screen (or triggers the activity re-creation in any other way) it'll navigate
+         * them to `My Tickets` again since the `originFromExtras` will still be [Origin.ZENDESK_NOTIFICATION].
+         */
+        if (savedInstanceState == null && originFromExtras == Origin.ZENDESK_NOTIFICATION) {
             showZendeskTickets()
         }
     }
@@ -96,11 +95,6 @@ class HelpActivity : AppCompatActivity() {
         refreshContactEmailText()
     }
 
-    override fun onSaveInstanceState(outState: Bundle?) {
-        outState?.putBoolean(HelpActivity.MY_TICKETS_SHOWN_KEY, mTicketsBeingShown)
-        super.onSaveInstanceState(outState)
-    }
-
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         if (item.itemId == android.R.id.home) {
             onBackPressed()
@@ -166,7 +160,6 @@ class HelpActivity : AppCompatActivity() {
     companion object {
         private const val ORIGIN_KEY = "ORIGIN_KEY"
         private const val EXTRA_TAGS_KEY = "EXTRA_TAGS_KEY"
-        private const val MY_TICKETS_SHOWN_KEY = "MY_TICKETS_SHOWN_KEY"
 
         @JvmStatic
         fun createIntent(
```

This greatly simplifies the solution by checking the nullity of `savedInstanceState` which tells us whether the activity is created for the first time or not. It also adds a comment explaining the process. I did double check that `savedInstanceState` is not null after re-creation even if we don't override the `onSaveInstanceState` and tested the steps [here](https://github.com/wordpress-mobile/WordPress-Android/pull/8054#discussion_r203149913).

Reply:
💯 addressed (by directly stealing your code) in 57b7759326